src/play_controller.hpp

```diff
@@ -132,10 +133,14 @@ class play_controller : public controller_base, public events::observer, public
 		gamestate().end_level_data_ = data;
 	}
 	void reset_end_level_data() {
-		gamestate().end_level_data_ = boost::none;
+		gamestate().end_level_data_.reset();
 	}
 	bool is_regular_game_end() const {
-		return gamestate().end_level_data_.get_ptr() != nullptr;
+#if defined HAVE_CXX17 || BOOST_VERSION >= 106800
```

Comment:
I realized I could have done `#ifdef HAVE_CXX17 ||` but is this ok?

Reply:
The parentheses are optional here, yes.